spec/models/record_spec.rb

```diff
@@ -1,13 +1,82 @@
+          "external_id": "link",
+          "title": "title",
+          "images": false
+        }
+      }
+    }
+    let(:feed_without_prefixed_date) {
+      {
+        "url": "https://www.seelow.de/news/rss.xml?rubrik=13",
+        "name": "Homepage Seelow",
+        "remove_prefixed_date_in_title": true,
+        "import": {
+          "intro": "description",
+          "body": false,
+          "external_id": "link",
+          "title": "title",
+          "images": false
+        }
+      }
+    }
+    let(:xml_data) {
+      # <?xml version="1.0"?>
+      # <rss>
+      #   <channel>
+      #     <item>
+      #       <title><![CDATA[19.10.2021: FACHFORUM "DASEINSVORSORGE" IM KREISKULTURHAUS]]></title>
+      #       <link>https://www.seelow.de/news/index.php?news=683934</link>
+      #       <description><![CDATA[<img src="https://vorschau.verwaltungsportal.de/news/6/8/3/9/3/4/3497999300.jpg" alt="FACHFORUM "DASEINSVORSORGE" IM KREISKULTURHAUS" /><br /><br /> Am 11.10.2021 kamen ca. 50 Referendare aus der ganzen Bundesrepublik zusammen und verständigten sich zum Thema &quot;Daseinsvorsorge im ländlichen Raum&quot;.
+      #         Nach Vorträgen u.a. von ... <br /><br />[<a href="https://www.seelow.de/news/index.php?news=683934" aria-label="Zur Meldung" title="Zur Meldung">mehr</a>]]]></description>
+      #       <pubDate>Tue, 19 Oct 2021 00:00:00 +0200</pubDate>
+      #     </item>
+      #   </channel>
+      # </rss>
+      Nokogiri::XML::Builder.new do |xml|
+        xml.rss {
+          xml.channel {
+            xml.item {
+              xml.title "19.10.2021: FACHFORUM \"DASEINSVORSORGE\" IM KREISKULTURHAUS"
+              xml.link "https://www.seelow.de/news/index.php?news=683934"
+              xml.description "<img src=\"https://vorschau.verwaltungsportal.de/news/6/8/3/9/3/4/3497999300.jpg\" alt=\"FACHFORUM \"DASEINSVORSORGE\" IM KREISKULTURHAUS\" /><br /><br /> Am 11.10.2021 kamen ca. 50 Referendare aus der ganzen Bundesrepublik zusammen und verständigten sich zum Thema &quot;Daseinsvorsorge im ländlichen Raum&quot;. Nach Vorträgen u.a. von ... <br /><br />[<a href=\"https://www.seelow.de/news/index.php?news=683934\" aria-label=\"Zur Meldung\" title=\"Zur Meldung\">mehr</a>"
+              xml.pubDate "Tue, 19 Oct 2021 00:00:00 +0200"
+            }
+          }
+        }
+      end
+    }
+
+    it "converts a xml from hash to json" do
+      xml_item = xml_data.doc.xpath("//item")
+      allow_any_instance_of(Record).to receive(:feed) { feed_with_prefixed_date }
+      expect(record.send(:parse_content_title, xml_item)).to eq("19.10.2021: FACHFORUM \"DASEINSVORSORGE\" IM KREISKULTURHAUS")
+    end
+
+    it "converts a xml from hash to json" do
+      xml_item = xml_data.doc.xpath("//item")
+      allow_any_instance_of(Record).to receive(:feed) { feed_without_prefixed_date }
+      expect(record.send(:parse_content_title, xml_item)).to eq("FACHFORUM \"DASEINSVORSORGE\" IM KREISKULTURHAUS")
+    end
```

Comment:
Personally, I would encapsulate the tests in `context`-blocks, and put better naming to the contexts and tests.
For example:

```ruby
describe "parse content title" do

  subject do
    allow_any_instance_of(Record).to receive(:feed) { feed }
    Record.new.send(:parse_content_title, xml_item) 
  end

  context "with prefixed date option enabled" do
    let(:feed) { "INSERT FEED WITH DATE OPTION HERE" }
  
    it "convertes xml to json and removes date" do
      expect(subject).to eq("STRING WITHOUT DATE")
    end
    
  context "without prefixed date option enabled" do
    let(:feed) { "INSERT FEED WITHOUT DATE OPTION HERE" }
  
    it "convertes xml to json" do
      expect(subject).to eq("STRING WITH DATE")
    end
  end
```

PS: I don't really understand why this option needs to be disabled, you could just look if there is a date and remove it, if there is one.
Also what is this feed used for, is it the general parsing options set for the whole project?

Reply:
>PS: I don't really understand why this option needs to be disabled, you could just look if there is a date and remove it, if there is one.
> Also what is this `feed` used for, is it the general parsing options set for the whole project?

it is set per importer source and can be activated for a particular one. we do not know if this is wanted in general.
the `feed` is one importer source. a project can have multiple `feed`s.
does this answer your question?

Reply:
yes.

I looked into the code now and since `feed` is just and `attr_accessor`, maybe you might as well do this since it's more explicit and less confusing:

```ruby
subject do
  record = Record.new
  record.feed = feed
  record.send(:parse_content_title, xml_item)
end
```

Reply:
We do not need `allow_any_instance_of` here, since we are only testing one instance.
This mocking was confusing to me.

Reply:
thanks, optimized in 25eab90